contracts/RootChain/RootChain.sol

```diff
@@ -201,6 +211,27 @@ contract RootChain {
         });
     }
 
+    /// For any attempted exit of an UTXO, validate that the UTXO's two inputs have not
+    /// been previously exited. If UTXO's inputs are in the exit queue, those inputs'
+    /// exits are deleted from the exit queue and the current UTXO's exit remains valid.
+    function validateExitInputs(RLP.RLPItem[] memory txList)
+        private
+    {
+        for (uint256 i = 0; i < 2; i++) {
+            uint256 txInputBlkNum = txList[5 * i + 0].toUint();
+            uint256 txInputIndex = txList[5 * i + 1].toUint();
+            uint256 txInputOutIndex = txList[5 * i + 2].toUint();
+            uint256 txInputPriority = 1000000000 * txInputBlkNum + 10000 * txInputIndex + txInputOutIndex;
+
+            require(finalizedExits[txInputPriority].owner == address(0));
+            require(finalizedExits[txInputPriority].amount == 0);
+
+            if (exits[txInputPriority].owner != address(0) || exits[txInputPriority].amount != 0) {
```

Comment:
this conditional does have to be here right? Already have the require statements before

Reply:
require statements check to see that the exit trying to be exited has not been finalized. If statement checks that it isn't currently trying to be exited (aka not finalized yet)

Reply:
oh shit sorry. eyes read the same thing